### PR TITLE
Update build to use NTL 11.5.1 and GMP 6.2.1

### PR DESCRIPTION
**Pull request by faberga**
Update build to use NTL 11.5.1 and GMP 6.2.1
Documentation update
Test docker containers update

### CODE REVIEW

**Comment by faberga** [comment]
Passed build and test on Ubuntu 20.04 ice-lake server and MacOS Catalina 10.15.7

File: docker/build_and_test/Dockerfile.FEDORA [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Recipe for an HE-ready Fedora 33 docker
```
Comment (jlhcrawford) [inline docker/build_and_test/Dockerfile.FEDORA]:
```suggestion
# Recipe for an HE-ready Fedora 34 docker
```

File: INSTALL.md [hunk 2]
```diff
@@ -216,14 +216,16 @@ step 3.
 **NOTE**: if linking against a non-system GMP, pass `GMP_PREFIX=<path/to/gmp>`
 to the `./configure` step.
 
-## Enabling and linking to Intel HEXL
+## Enabling and linking to Intel® HEXL
+**NOTE:** HEXL is only supported on the processors with AVX512DQ and AVX512-IFMA
```
Comment (jlhcrawford) [inline INSTALL.md]: This is not strictly true. It is *recommended* to have AVX512 support but HEXL does work without it.

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,13 @@
-FROM ubuntu:18.04
+FROM ubuntu:20.04
```
Comment (jlhcrawford) [inline Dockerfile]: What even is the purpose of this file if there is an Ubuntu 20.04 docker file in `docker/build_and_test` [here](https://github.com/helibproject/HElib/blob/master/docker/build_and_test/Dockerfile.UBUNTU)?

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 
 COPY ./ /root/HElib/
 
 RUN apt update && \
     apt install -y build-essential wget git cmake m4 libgmp-dev file && \
     cd ~ && \
-    wget https://www.shoup.net/ntl/ntl-11.4.1.tar.gz && \
-    tar --no-same-owner -xf ntl-11.4.1.tar.gz && \
-    cd ntl-11.4.1/src && \
+    wget https://libntl.org/ntl-11.5.1.tar.gz && \
```
Comment (jlhcrawford) [inline Dockerfile]: On Ubuntu 20.04 you don't need to download from source. You can do `apt-get install libntl-dev`.

File: docker/build_and_test/Dockerfile.FEDORA (hunk 1 above)
Comment (jlhcrawford) [reply docker/build_and_test/Dockerfile.FEDORA]: Changed comment to match version pulled by docker.

File: Dockerfile (hunk 4 above)
Comment (hamishun) [reply Dockerfile]: support for latest NTL is not always available via apt.

File: INSTALL.md (hunk 2 above)
Comment (faberga) [reply INSTALL.md]: I can change to HEXL **_acceleration_** is only supported ... with AVX512.

Without AVX512 it uses the reference implementation in C++ with no acceleration. in fact slows down when compared with NTL.
if we don't say anything, we would be misleading people.

File: Dockerfile (hunk 3 above)
Comment (faberga) [reply Dockerfile]: there was an original dockerfile in the top directory .... I can just delete it.

File: Dockerfile (hunk 4 above)
Comment (faberga) [reply Dockerfile]: On ubunt 20.04, "apt"  installs 11.4.3 and not 11.5.1 !

File: INSTALL.md (hunk 2 above)
Comment (faberga) [reply INSTALL.md]: Plus, this is a statement about HElib with HEXL acceleration, and not about HEXL.

File: INSTALL.md [hunk 5]
```diff
@@ -216,14 +216,16 @@ step 3.
...
+such as the 3rd generation Intel® Xeon® or the 11th generation Intel® Core®
```
Comment (faberga) [inline INSTALL.md]:
```suggestion
**NOTE:** HElib with HEXL acceleration is only supported on the processors with AVX512DQ and 
AVX512-IFMA such as the 3rd generation Intel® Xeon® or the 11th generation Intel® Core®
```

File: Dockerfile (hunk 3 above)
Comment (jlhcrawford) [reply Dockerfile]: I don't think it is necessary anymore unless it is doing something that the one in `docker/build_and_test` cannot do.

File: Dockerfile (hunk 4 above)
Comment (jlhcrawford) [reply Dockerfile]: Fair enough. We should still ideally be testing with the "default" version that comes with apt.

**Review by jlhcrawford — APPROVED** [review]

**Pull request merged**